workerapi/worker.thrift

```diff
@@ -0,0 +1,38 @@
+enum Status {
+  UNKNOWN = 0
+  READY = 1
+  PENDING = 2
+  FAILURE = 3
+  RUNNING = 4
+  SUCCESS = 5
+  ABORTED = 6
+  TIMEOUT = 7
+}
+
+struct WorkerStatus {
+  1: required Status status
+  2: optional string taskId
+  3: optional string outUri
+  4: optional string errUri
+  5: optional string info
+  6: optional i32 exitCode
+}
+
+struct Task {
+  1: required list<string> argv
+  2: optional list<string> env
+  3: optional string dir
+  4: optional string taskId
```

Comment:
Yes, the best way to do that is to have the Worker maintain its own ID space.

Let me give you another example:
the Scheduler schedules task <id1> on worker 1.
The task fails, and the Scheduler tries to reschedule it on both worker 1 and worker 2.

Then worker1 has seen the same id twice. This is confusing.

Also confusing is if there are two schedulers.

This is confusing because we are conflating the desire (run this Task) and the action (one Run of the Task). We may want to Run one Task multiple times.

I think the way to make this simplest is to have the Worker interface assign a new ID each time we start a Run.

Yes, that means it's up to the Scheduler to maintain a mapping of Task -> List<Run>.

BUT, that's also a *good* thing: the Scheduler needs to know to not rerun the Task if it's already failed N times!

WDYT?